### PR TITLE
Feature - Add support for outbound webhooks

### LINKED ISSUE #137: Feature - Support creation/deletion of Webhooks
It is now possible to create [outbound webhooks](https://docs.getdbt.com/docs/deploy/webhooks) in dbt Cloud.
The provider could be used to create those.

I am working on it and will have a PR in the next few days.

### PR DESCRIPTION
**Pull request by b-per**
Fixes #137

Will most likely conflict on `provider.go` with the PR for the service token.
We will also need to know if we want to merge #133 before or after as this will require som slight docs rework

### datasource acceptance test passing
SCR-20230517-qikz

### resource acceptance test passing
SCR-20230517-rxde

### CODE REVIEW

**Comment by b-per** [comment]
The CI issues seems to be about a problem connecting to dbt Cloud and not an issue with the new tests.

File: pkg/dbt_cloud/webhook.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	name string,
+	description string,
+	clientUrl string,
+	eventTypes []string,
+	jobIds []int,
+	active bool,
+) (*WebhookRead, error) {
+
+	newWebhook := WebhookWrite{
+		WebhookId:   webhookId,
```
Comment (GtheSheep) [inline pkg/dbt_cloud/webhook.go:70]: shouldn't this not exist until after creation?

File: pkg/resources/webhook.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+		if d.HasChange("job_ids") {
+			jobIDs := make([]int, len(d.Get("job_ids").([]interface{})))
+			for i, step := range d.Get("job_ids").([]interface{}) {
+				jobIDs[i] = step.(int)
+			}
+			webhook.JobIds = jobIDs
+		}
+		if d.HasChange("active") {
+			active := d.Get("active").(bool)
+			webhook.Active = active
```
Comment (GtheSheep) [inline pkg/resources/webhook.go:257]: just to check, this doesn't delete it if it's set to not active like some of the other endpoints, right?

File: pkg/resources/webhook.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "Webhooks active flag",
+	},
+	"hmac_secret": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Sensitive:   true,
+		Description: "Webhooks active flag",
```
Comment (GtheSheep) [inline pkg/resources/webhook.go:70]: wrong desc I think?

File: pkg/resources/webhook.go [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+	},
+	"http_status_code": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Webhooks HTTP Status Code",
+	},
+	"account_identifier": &schema.Schema{
```
Comment (GtheSheep) [inline pkg/resources/webhook.go:77]: is this the dbt Cloud account ID? Think it exists on the Client which could save having it here?

File: examples/resources/dbt_cloud_webhook/resource.tf [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  client_url = "http://localhost/nothing"
+  event_types = [
+    "job.run.started",
+    "job.run.completed"
+  ]
+  job_ids = [
+    1234,
+    5678
+  ]
+}
```
Comment (GtheSheep) [inline examples/resources/dbt_cloud_webhook/resource.tf:13]: nit: New line at EOF

File: pkg/resources/webhook.go (hunk 3 above)
Comment (b-per) [reply pkg/resources/webhook.go:70]: Indeed, some copy/paste issue. I will update it.

File: pkg/dbt_cloud/webhook.go (hunk 1 above)
Comment (b-per) [reply pkg/dbt_cloud/webhook.go:70]: It is actually required but with an empty value for the creation of the webhook.

It is set up empty [here](https://github.com/b-per/terraform-provider-dbt-cloud/blob/2e132ec6690c6b8b3f57a4add47de3e98dff52be/pkg/resources/webhook.go#L155)

File: pkg/resources/webhook.go (hunk 2 above)
Comment (b-per) [reply pkg/resources/webhook.go]: The API doesn't actually allow setting active to `false` and we need to `DELETE` it instead. I remove this from triggering a change to the resource.

**Review by GtheSheep — APPROVED** [review]

**Pull request merged**